controllers/ArticleController.js

```diff
@@ -0,0 +1,237 @@
+          });
+        }
+
+        return res.status(200).json({ article });
+      });
+  }
+
+  /**
+   * get all articles created
+   * @param {object} req - request object
+   * @param {object} res - response object
+   * @param {object} next: response object
+   * @returns {object} - the found article from database or empty if not found
+   */
+  static listAllArticles(req, res, next) {
+    return Article
+      .findAll({
+        include: [{
+          model: User,
+          attributes: { exclude: ['id', 'email', 'hashedPassword', 'createdAt', 'updatedAt'] }
+        }],
+        attributes: { exclude: ['id', 'userId'] }
+      })
+      .then((articles) => {
+        /** check if there was no article created */
+        if (articles.length === 0) {
+          return res.status(200).json({
+            message: 'Your request was successful but no articles created',
+            articles,
+          });
+        }
+
+        return res.status(200).json({ articles, articlesCount: articles.length });
+      })
+      .catch(next);
+  }
+
+  /**
+   * @function editArticle
+  * @summary: API controller to handle requests
+  * to edit an article
+  * @param {object} req: request object
+  * @param {object} res: response object
+  * @param {object} next: response object
+  * @returns {object} api response: article object for
+  * successful requests, or error object for
+  * requests that fail
+  */
+  static editArticle(req, res, next) {
+    const { title, description, body } = req.body.article;
+    const { count } = req;
+    const { slug } = req.params;
+    return Article.update({
+      title,
+      description,
+      body,
+      updatedCount: Utilities.increaseCount(count)
+    }, {
+      where: {
+        slug,
+      },
+      returning: true,
+      plain: true
+    })
+      .then(result => res.status(200).json({
+        success: true,
+        article: result[1]
+      }))
+      .catch(next);
+  }
+
+  /**
+ * @function deleteArticle
+* @summary: API controller to handle requests
+* to delete an article
+* @param {object} req: request object
+* @param {object} res: response object
+* @param {object} next: response object
+* @returns {object} api response: article object for
+* successful requests, or error object for
+* requests that fail
+*/
+  static deleteArticle(req, res, next) {
+    const { slug } = req.params;
+    Article.destroy({
+      where: { slug }
+    })
+      .then(() => res.status(204).json())
+      .catch(next);
+  }
+
+  /**
+ * @function likeArticle
+* @summary: API controller to handle requests
+* to like an article
+* @param {object} req: request object
+* @param {object} res: response object
+* @param {object} next: response object
+* @returns {object} api response: article object for
+* successful requests, or error object for
+* requests that fail
+*/
+  static likeArticle(req, res, next) {
+    const { userId } = req;
+
+    const { slug } = req.params;
+    let articleId, likesCount;
```

Comment:
I think this likesCount implementation has to be checked because while it works on development, it breaks on production.